test/e2e/suite_test.go

```diff
@@ -0,0 +1,331 @@
+					"install",
+					"--set",
+					"bootstrapPassword=admin",
+					"--set",
+					"replicas=1",
+					"--set", fmt.Sprintf("hostname=%s.%s", e2eCfg.ExternalIP, e2eCfg.MagicDNS),
+					"--create-namespace",
+					rancherName,
+					fmt.Sprintf(e2eCfg.RancherChartURL),
+				)).To(Succeed())
+				Eventually(func() bool {
+					return isDeploymentReady(cattleSystemNamespace, rancherName)
+				}, 5*time.Minute, 2*time.Second).Should(BeTrue())
+			}
+		})
+	})
+
+	By("Deploying eks operator CRD chart", func() {
+		if isDeploymentReady(cattleSystemNamespace, operatorName) {
+			By("already installed")
+		} else {
+			Expect(kubectl.RunHelmBinaryWithCustomErr(
+				"-n",
+				crdChartName,
+				"install",
+				"--create-namespace",
+				"--set", "debug=true",
+				operatorName,
+				e2eCfg.CRDChart,
+			)).To(Succeed())
+
+			By("Waiting for CRDs to be created")
+			Eventually(func() bool {
+				for _, crdName := range crdNames {
+					crd := &apiextensionsv1.CustomResourceDefinition{}
+					if err := cl.Get(ctx,
+						runtimeclient.ObjectKey{
+							Name: crdName,
+						},
+						crd,
+					); err != nil {
+						return false
+					}
+				}
+				return true
+			}, 5*time.Minute, 2*time.Second).Should(BeTrue())
+		}
+	})
+
+	By("Deploying eks operator chart", func() {
+		if isDeploymentReady(cattleSystemNamespace, operatorName) {
+			By("already installed")
+		} else {
+			Expect(kubectl.RunHelmBinaryWithCustomErr(
+				"-n",
+				cattleSystemNamespace,
+				"install",
+				"--create-namespace",
+				"--set", "debug=true",
+				operatorName,
+				e2eCfg.OperatorChart,
+			)).To(Succeed())
+
+			By("Waiting for eks operator deployment to be available")
+			Eventually(func() bool {
+				return isDeploymentReady(cattleSystemNamespace, operatorName)
+			}, 5*time.Minute, 2*time.Second).Should(BeTrue())
+		}
+		// As we are not bootstrapping rancher in the tests (going to the first login page, setting new password and rancher-url)
+		// We need to manually set this value, which is the same value you would get from doing the bootstrap
+		setting := &managementv3.Setting{}
+		Expect(cl.Get(ctx,
+			runtimeclient.ObjectKey{
+				Name: "server-url",
+			},
+			setting,
+		)).To(Succeed())
+
+		setting.Source = "env"
+		setting.Value = fmt.Sprintf("https://%s.%s", e2eCfg.ExternalIP, e2eCfg.MagicDNS)
+
+		Expect(cl.Update(ctx, setting)).To(Succeed())
+
+	})
+
+	By("Creating aws credentials secret", func() {
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      awsCredentialsSecretName,
+				Namespace: "default",
+			},
+			Data: map[string][]byte{
+				"amazonec2credentialConfig-accessKey": []byte(e2eCfg.AWSAccessKey),
+				"amazonec2credentialConfig-secretKey": []byte(e2eCfg.AWSSecretAccessKey),
+			},
+		}
+
+		err := cl.Create(ctx, secret)
+		if err != nil {
+			fmt.Println(err)
+			Expect(apierrors.IsAlreadyExists(err)).To(BeTrue())
+		}
+	})
+
+	// By("Reading cluster templates", func() { TODO: uncomment with first tests
```

Comment:
nit: do we need this or can we delete it?

Reply:
We need it, I will uncomment this with the first tests.